pylon-core/src/main/kotlin/io/github/pylonmc/pylon/core/persistence/Serializers.kt

```diff
@@ -67,3 +86,155 @@ class UUIDPersistentDataType : PersistentDataType<LongArray, UUID> {
+
+    override fun fromPrimitive(primitive: ByteArray, context: PersistentDataAdapterContext): World {
+        val uid = Serializers.UUID.fromPrimitive(primitive, context)
+        return Bukkit.getWorld(uid) ?: throw IllegalArgumentException(uid.toString())
+    }
+
+    override fun toPrimitive(complex: World, context: PersistentDataAdapterContext): ByteArray {
+        return Serializers.UUID.toPrimitive(complex.uid, context)
+    }
+}
+
+class LocationPersistentDataType : PersistentDataType<ByteArray, Location> {
+    override fun getPrimitiveType(): Class<ByteArray>
+            = ByteArray::class.java
+
+    override fun getComplexType(): Class<Location>
+            = Location::class.java
+
+    override fun fromPrimitive(primitive: ByteArray, context: PersistentDataAdapterContext): Location {
+        val buffer = ByteBuffer.wrap(primitive)
+        val world = Serializers.WORLD.fromPrimitive(primitive, context)
+        val x = buffer.getDouble()
+        val y = buffer.getDouble()
+        val z = buffer.getDouble()
+        val yaw = buffer.getFloat()
+        val pitch = buffer.getFloat()
+        return Location(world, x, y, z, yaw, pitch)
+    }
+
+    override fun toPrimitive(complex: Location, context: PersistentDataAdapterContext): ByteArray {
+        val buffer = ByteBuffer.allocate(2 * Long.SIZE_BYTES + 3 * Double.SIZE_BYTES + 2 * Float.SIZE_BYTES)
+        buffer.put(Serializers.WORLD.toPrimitive(complex.world, context))
+        buffer.putDouble(complex.x)
+        buffer.putDouble(complex.y)
+        buffer.putDouble(complex.z)
+        buffer.putFloat(complex.yaw)
+        buffer.putFloat(complex.pitch)
+        return buffer.array()
+    }
+}
+
+class BlockPositionPersistentDataType : PersistentDataType<ByteArray, BlockPosition> {
+    override fun getPrimitiveType(): Class<ByteArray>
+            = ByteArray::class.java
+
+    override fun getComplexType(): Class<BlockPosition>
+            = BlockPosition::class.java
+
+    override fun fromPrimitive(primitive: ByteArray, context: PersistentDataAdapterContext): BlockPosition {
+        val buffer = ByteBuffer.wrap(primitive)
+        var world: World? = null
+        if(buffer.get() == 1.toByte()){
+            val mostSignificantBits = buffer.getLong()
+            val leastSignificantBits = buffer.getLong()
+            world = Bukkit.getWorld(UUID(mostSignificantBits, leastSignificantBits))
+        }
+        val x = buffer.getInt()
+        val y = buffer.getInt()
+        val z = buffer.getInt()
+        return BlockPosition(world, x, y, z)
+    }
+
+    override fun toPrimitive(complex: BlockPosition, context: PersistentDataAdapterContext): ByteArray {
+        val buffer = ByteBuffer.allocate(2 * Long.SIZE_BYTES + 3 * Int.SIZE_BYTES + 1)
+        val world = complex.world
+        if(world != null){
+            buffer.put(1)
+            buffer.put(Serializers.WORLD.toPrimitive(complex.world!!, context))
+        }
+        buffer.put(0)
```

Comment:
```suggestion
        if(world != null){
            buffer.put(1)
            buffer.put(Serializers.WORLD.toPrimitive(complex.world!!, context))
        } else {
            buffer.put(0)
        }
```
Data corruption possibility